packages/rum/src/domain/getSessionReplayLink.ts

```diff
@@ -0,0 +1,50 @@
+import type { RumConfiguration, RumSessionManager, ViewContexts } from '@datadog/browser-rum-core'
+import { getDatadogOrigin } from '@datadog/browser-rum-core'
+import { isBrowserSupported } from '../boot/isBrowserSupported'
+import { getReplayStats } from './replayStats'
+
+export function getSessionReplayLink(
+  configuration: RumConfiguration,
+  sessionManager: RumSessionManager,
+  viewContexts: ViewContexts
+): string | undefined {
+  const session = sessionManager.findTrackedSession()
+  const queryParams: Record<string, string> = {}
+  let sessionId: string
+  if (session) {
+    sessionId = session.id
+
+    if (session.sessionReplayAllowed === false) {
+      // possibilities
+      // - replay sampled out
+      queryParams['error-type'] = 'incorrect-session-plan'
+    }
+  } else {
+    // possibilities:
+    // - rum sampled out
+    // - session expired (edge case)
+    queryParams['error-type'] = 'rum-not-tracked'
+    sessionId = 'session-id'
+  }
+  const view = viewContexts.findView()
+  if (view) {
+    queryParams['seed'] = view.id
+    queryParams['from'] = `${view.startClocks.timeStamp}`
+    const replayStats = getReplayStats(view.id)
+    if (!replayStats && !queryParams['error-type']) {
+      queryParams['error-type'] = 'replay-not-started'
+    }
+  }
+  if (!isBrowserSupported()) {
+    queryParams['error-type'] = 'browser-not-supported'
+  }
+
+  let path = `/rum/replay/sessions/${sessionId}`
+  if (Object.keys(queryParams).length > 0) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I don't think we can have zero param in practice? I think even if we can, it's fine to have the question mark in this case, so the code could look like this:

```ts
return `${origin}${path}?${parameters.join('&')}`
```